pybossa/view/projects.py

```diff
@@ -1698,6 +1700,106 @@ def bulk_priority_update(short_name):
+@login_required
+@admin_or_subadmin_required
+def bulk_update_assign_worker(short_name):
+   
+    response = {}
+    project, owner, ps = project_by_shortname(short_name)
+    data = json.loads(request.data)
+
+    if data.get("add") is None and data.get("remove") is None:
+        # read data and return users
+
+        task_id = data.get("taskId")
+        bulk_update = False
+        assign_user_emails = []
+        if task_id:
+            # use filters tp populate user list
+            t = task_repo.get_task_by(project_id=project.id,
+                                        id=int(task_id))
+            assign_user_email = []
+            if t.user_pref is not None and isinstance(t.user_pref, dict):
+                assign_user_emails = set(t.user_pref.get("assign_user", []))
+        else:
+            bulk_update = True
+            args = parse_tasks_browse_args(json.loads(data.get('filters', 'null')))
+            tasks = task_repo.get_tasks_by_filters(project, args)
+            task_ids = [t.id for t in tasks]
+            assign_user_emails = set()
+
+            for task_id in task_ids:
+                t = task_repo.get_task_by(project_id=project.id,
+                                        id=int(task_id))
+                assign_user_emails = assign_user_emails.union(set(t.user_pref.get("assign_user", [])))
+        assign_users = []
+        for user_email in assign_user_emails:
+            user = user_repo.search_by_email(user_email)
+            if not user:
+                fullname = user_email + ' (user not found)'
+            else:
+                fullname = user.fullname 
+            assign_users.append({'fullname': fullname, 'email': user_email})
+        response['assign_users'] = assign_users
+
+        # get a list of all users can be assigned to task
+        if bool(data_access_levels):
+            all_users = user_repo.get_users(project.get_project_users())
+        else:
+            all_users = user_repo.get_all()
+        all_user_data = []
+        for user in all_users:
+
+            # Exclude currently assigned users in the candidate list ONLY for single task update
+            if user.email_addr in assign_user_emails and not bulk_update:
+                continue
+            user_data = dict()
+            user_data['fullname'] = user.fullname
+            user_data['email'] = user.email_addr
+            all_user_data.append(user_data)
+        response["all_users"] = all_user_data
+    else:
+        # update tasks with assign worker values
+        assign_workers = data.get('add', [])
+        remove_workers = data.get('remove', [])
+
+        assign_worker_emails = [w["email"] for w in assign_workers]
+        remove_worker_emails = [w["email"] for w in remove_workers]
+
+        task_id = data.get("taskId")
+
+        if not task_id:
+            # get task_ids from db
+            args = parse_tasks_browse_args(json.loads(data.get('filters', '')))
```

Comment:
Similar to the above comment and [solution](https://github.com/bloomberg/pybossa/pull/655/files#r771642803) recommend a check for None here.

Also, should the default value be `''` or `null` as in the prior code block above?